internal/services/dataprotection/data_protection_backup_policy_postgresql_flexible_server_resource_test.go

```diff
@@ -0,0 +1,205 @@
+	})
+}
+
+func (r DataProtectionBackupPolicyPostgreSQLFlexibleServerResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := backuppolicies.ParseBackupPolicyID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.DataProtection.BackupPolicyClient.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving DataProtection BackupPolicy (%q): %+v", id, err)
+	}
+	return utils.Bool(true), nil
+}
+
+func (r DataProtectionBackupPolicyPostgreSQLFlexibleServerResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-dataprotection-%d"
+  location = "%s"
+}
+
+resource "azurerm_data_protection_backup_vault" "test" {
+  name                = "acctest-dbv-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  datastore_type      = "VaultStore"
+  redundancy          = "LocallyRedundant"
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func (r DataProtectionBackupPolicyPostgreSQLFlexibleServerResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_protection_backup_policy_postgresql_flexible_server" "test" {
+  name                            = "acctest-dbp-%d"
+  vault_id                        = azurerm_data_protection_backup_vault.test.id
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
+
+  default_retention_rule {
+    life_cycle {
+      duration        = "P4M"
+      data_store_type = "VaultStore"
+    }
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r DataProtectionBackupPolicyPostgreSQLFlexibleServerResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_protection_backup_policy_postgresql_flexible_server" "import" {
+  name                            = azurerm_data_protection_backup_policy_postgresql_flexible_server.test.name
+  vault_id                        = azurerm_data_protection_backup_policy_postgresql_flexible_server.test.vault_id
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
+
+  default_retention_rule {
+    life_cycle {
+      duration        = "P4M"
+      data_store_type = "VaultStore"
+    }
+  }
+}
+`, r.basic(data))
+}
+
+func (r DataProtectionBackupPolicyPostgreSQLFlexibleServerResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_data_protection_backup_policy_postgresql_flexible_server" "test" {
+  name                            = "acctest-dbp-%d"
+  vault_id                        = azurerm_data_protection_backup_vault.test.id
+  backup_repeating_time_intervals = ["R/2021-05-23T02:30:00+00:00/P1W"]
```

Comment:
shall we add a testcase for `backup_repeating_time_intervals ` with more than one element?

Reply:
Updated